README.md

```diff
@@ -1,187 +1,173 @@
-        // do some validation here
-    }
-
-    if err != nil {
-        return nil, err
-    }
+resources, err := keptnAPI.ResourcesV1().GetAllServiceResources("my-project", "my-stage", "my-service")
+if err != nil {
+log.Fatal(err)
 }
-
 ```
 
-### CloudEvent Data
-If you need to access data within CloudEvents:
+### Accessing the Keptn API from within the control plane
+
+If you are developing a Keptn integration that is supposed to run *inside/as part of* the Keptn control plane,
+there is no need to talk to the Keptn API via the API gateway. In this case one option is to use the `InternalAPISet`.
+
+Import the following package:
 
 ```go
-import {
-	"github.com/keptn/go-utils/pkg/lib"
-)
+import api "github.com/keptn/go-utils/pkg/api/utils"
 ```
 
-Example:
+Then create an `InternalAPISet`. Note, that this does *not* require you to pass the URL to the keptn API or an API token
 
 ```go
-func parseCloudEvent(event cloudevents.Event) (keptnevents.TestFinishedEventData, error) {
-	eventData := &keptn.TestsFinishedEventData{}
-	err := event.DataAs(eventData)
-    
-    return eventData, err
+keptnAPI, err := api.NewInternal(nil)
+if err != nil {
+log.Fatal(err)
 }
 ```
 
-### Models
-If you need to access Models for YAML files:
+## Creating and inspecting Keptn events:
 
 ```go
-import {
-	"github.com/keptn/go-utils/pkg/lib"
-)
+// Creating a new event
+newEvent, _ := lib.KeptnEvent(lib.GetStartedEventType("echo-task"), "my-service", lib.EventData{
+Project: "my-project",
+Stage:   "my-stage",
+Service: "my-service",
+}).Build()
+
+// Converting KeptnContextExtendedCE to cloudevents SDK event
+cloudEvent := lib.ToCloudEvent(newEvent)
+fmt.Println(cloudEvent.ID())
+
+// Converting cloudevents SDK event to KeptnContextExtendedCE
+newEvent, _ = lib.ToKeptnEvent(cloudEvent)
+
+// Marshalling a keptn event to JSON
+newEventAsJSON, _ := newEvent.ToJSON()
+fmt.Println(string(newEventAsJSON))
+
+// Creating a started event for a triggered event
+startedEventType, _ := lib.ReplaceEventTypeKind(*triggeredEvent.Type, "started")
+startedEvent := lib.KeptnEvent(startedEventType, "my-service", triggeredEvent.Data)
+
+startedEventAsJSON, _ := startedEvent.ToJSON()
+fmt.Println(string(startedEventAsJSON))
+
+// Checking for event types
+fmt.Println(lib.IsSequenceEventType("sh.keptn.event.echo.triggered")) // false
+fmt.Println(lib.IsSequenceEventType("sh.keptn.event.stage.echo.triggered")) // true
+fmt.Println(lib.IsTaskEventType("sh.keptn.event.echo.triggered")) // true
+fmt.Println(lib.IsTaskEventType("sh.keptn.event.stage.echo.triggered")) // false
+fmt.Println(lib.IsTriggeredEventType("sh.keptn.event.echo.triggered")) // true
 ```
 
-### Querying Events from event store
-```
-// Create an event Handler
-eventHandler := apiutils.NewAuthenticatedEventHandler("1.2.3.4/api", token, "x-token", nil, "http")
+## Querying Keptn events
+
+```go
+// Creating the APISet
+apiSet, _ := api.New("http://<keptn-url>/api", api.WithAuthToken("<api-token>"))
+
```

Comment:
here we should mention that for the `EventFilter` either `Project` or `KeptnContext` has to be set for the query to work properly

Reply:
You mean that it will not work if both are set? or what does "work properly" mean?

Reply:
setting both will work, but at least one of them needs to be set